### PR TITLE
[MRESOLVER-232] Refactor selector

### PR DESCRIPTION
**Pull request by cstamas**
Make SimpleNamedLockFactorySelector logic reusable

### CODE REVIEW

**Comment by michael-o** [comment]
Need to take a look at this with a sober mind.

**Comment by michael-o** [comment]
Maybe this will be simpler when service locator will be gone?

**Comment by cstamas** [comment]
> Maybe this will be simpler when service locator will be gone?

Def, so can't wait to drop SL

**Comment by michael-o** [comment]
> > Maybe this will be simpler when service locator will be gone?
> 
> Def, so can't wait to drop SL

Does this make sense while SL is still there?

File: maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/synccontext/named/NamedLockFactorySelectorSupport.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+     */
+    @Override
+    public NameMapper getSelectedNameMapper()
+    {
+        return nameMapper;
+    }
+
+    /**
+     * Returns selected factory name (or sane default) using System property value of {@link #FACTORY_KEY} and defaults
+     * to {@link LocalReadWriteLockNamedLockFactory#NAME}.
```
Comment (pzygielo) [inline maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/synccontext/named/NamedLockFactorySelectorSupport.java:73]: Here the default actually used is out of control of this class, so this comment does not seem right.

Comment (cstamas) [reply maven-resolver-impl/src/main/java/org/eclipse/aether/internal/impl/synccontext/named/NamedLockFactorySelectorSupport.java]: Fixed, thanks!

**Comment by cstamas** [comment]
> Does this make sense while SL is still there?

IMHO yes

**Review by michael-o — CHANGES_REQUESTED** [review]
I think this deserves a JIRA issue with a description who would benefit from. The change itself is fine by me.

**Comment by jira-importer** [comment]
Resolve #944

**Pull request merged**